src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -464,85 +465,92 @@ protected internal virtual DurableOrchestrationClient GetClient(OrchestrationCli
             };
         }
 
-        internal void RegisterOrchestrator(FunctionName orchestratorFunction, OrchestratorInfo orchestratorInfo)
+        internal void RegisterOrchestrator(FunctionName orchestratorFunction, RegisteredFunctionInfo orchestratorInfo)
         {
-            if (!this.registeredOrchestrators.TryUpdate(orchestratorFunction, orchestratorInfo, null))

```

Comment:
I am very confused by the old logic here. How is `TryUpdate(orchestratorFunction, orchestratorInfo, null)` different than `TryAdd(orchestratorFunction, orchestratorInfo)`?

In regards to the new logic, I am a little confused why we use `TryAdd()` with a `[]` update if we couldn't add. `TryAddOrUpdate` seems to accomplish the same thing.  Is this so that we only log in the case where we are adding a new orchestrator as opposed to updating an existing one?

Reply:
I was confused about this too, but I think what happened. Previously this method was implemented the same way as `RegisterActivity`, and that use of `TryUpdate` actually made sense. Katy later made a change where instead of `ITriggeredFunctionExecutor` we used `OrchestratorInfo` and I think that caused this old logic to no longer work the way it was intended.

The use of `[]` over `TryAddOrUpdate` was somewhat arbitrary. I felt the behavior was properly understood and less verbose, which is why I went with it.

Reply:
Oh, I might have misuderstood your second question. The reason I do TryAdd followed by a direct `[]` update is because I want to distinguish between add and update, basically so I can write the trace only in the add case.